### PR TITLE
ENH: Add numba engine to groupby.var/std

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -1272,6 +1272,7 @@ def _numba_agg_general(
         func: Callable,
         engine_kwargs: dict[str, bool] | None,
         numba_cache_key_str: str,
+        *aggregator_args,
```
Comment (jreback) [inline pandas/core/groupby/groupby.py:1275]: why are these not kwargs?

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -2000,23 +2006,52 @@ def std(self, ddof: int = 1):
...
             Degrees of freedom.
 
+        engine : str, default None
+            * ``'cython'`` : Runs the operation through C-extensions from cython.
+            * ``'numba'`` : Runs the operation through JIT compiled code from numba.
+            * ``None`` : Defaults to ``'cython'`` or globally setting
+              ``compute.use_numba``
+
+            .. versionadded:: 1.4.0
+
```
Comment (jreback) [inline pandas/core/groupby/groupby.py:2016]: in a followup can you try to use shared docstrings here (and the other functions) for the engine/engine_kwargs

**Review by jreback — APPROVED** [review]
thanks @mroeschke followup request

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/groupby/groupby.py:1275]: Numba functions don't play nice with kwargs currently https://github.com/numba/numba/issues/2916

File: pandas/core/groupby/groupby.py (hunk 2 above)
Comment (mroeschke) [reply pandas/core/groupby/groupby.py:2016]: Sure I can try to upstream these doc parameters from rolling too

**Pull request merged**